src/main/java/org/jenkinsci/plugins/workflow/libs/GlobalUntrustedLibraries.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * The MIT License
+ *
+ * Copyright 2024 CloudBees, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+package org.jenkinsci.plugins.workflow.libs;
+
+import edu.umd.cs.findbugs.annotations.NonNull;
+import hudson.Extension;
+import hudson.ExtensionList;
+import hudson.security.Permission;
+import jenkins.model.Jenkins;
+
+/**
+ * Manages untrusted libraries available to any job in the system.
+ */
+@Extension public class GlobalUntrustedLibraries extends AbstractGlobalLibraries {
+
+    public GlobalUntrustedLibraries() {
+        super();
+    }
+
+    @Override
+    public String getDescription() {
+        return Messages.GlobalUntrustedLibraries_Description();
+    }
+
+    @NonNull
+    @Override
+    public String getDisplayName() {
+        return Messages.GlobalUntrustedLibraries_DisplayName();
+    }
+
+    public static @NonNull GlobalUntrustedLibraries get() {
+        return ExtensionList.lookupSingleton(GlobalUntrustedLibraries.class);
+    }
+
+    @NonNull
+    @Override
+    public Permission getRequiredGlobalConfigPagePermission() {
+        return Jenkins.MANAGE;
+    }
+
+    @Extension(ordinal=0) public static class ForJob extends AbstractForJob {
```

Comment:
I think this implementation needs to wrap the returned libraries in `ResolvedLibraryConfiguration` like `FolderLibraries.ForJob` to avoid introducing a new security issue like what was fixed by https://github.com/jenkinsci/pipeline-groovy-lib-plugin/commit/ace0de3c2d691662021ea10306eeb407da6b6365.

Reply:
Maybe the trust flag is enough though, I am not sure. Either way I would look at old security fixes here carefully to see if anything extra needs to be done to separate the two global configurations in all contexts.

Reply:
Indeed, this is addressed in 98322fb

Reply:
Looking at the old fix in more details, I think it was probably fine as you had it originally because of this line: https://github.com/jenkinsci/pipeline-groovy-lib-plugin/blob/b98828a1b67b347010d3cf497d68fe4b193291b8/src/main/java/org/jenkinsci/plugins/workflow/libs/LibraryAdder.java#L132

I think folder-based libraries just needed special treatment to distinguish between folders with the same name defined at different levels since they use the same `LibraryResolver` class. Since the new untrusted global libraries use a distinct resolver class and only exist in a single place, things should be fine without wrapping `LibraryConfiguration`s. Your changes use `getClass().getName()` anyway so the behavior should be identical. I didn't try testing it though.